blreams/collegefootballpick10
# Enter picks Pull Request

This pull request implements an "enter picks page" and it also reworked the navigation links at the top of the body to allow easier navigation between pages (this was done specifically to get to the enter picks page).  Sorry that this post went a little long.

**Enter Picks Page**
```
# the following requirements are required to get the page to show up
1.  A user must be logged in
2.  The user must be linked to a player in the year being requested
3.  It must be before the pick deadline or lock_picks should be None
4.  The week must be in the NOT_STARTED state
```
```Python
# major files
pick10/enter_picks_view.py   # renders the enter picks view
pick10/enter_picks.py   # get picks and save picks functions
templates/pick10/enter_picks.html   # the enter pick html template
functional_tests/test_enter_picks.py  # extensive testing
```
*A few comments*
* I'm not entirely satisfied with the user interface yet.  If the screen is wide enough I think it looks ok
* Using the enter_picks webpage to submit picks is extensively tested
* If someone writes a program to make a POST request to set picks, this was not tested.  There are protections in the POST method to prevent malicious POST requests, but they are untested.
* Writing the picks to an excel spreadsheet has not been implemented yet
* I didn't follow the django forms, I would need to research this more to figure out how to use it

*If you want to test out the page*
* If have the year 2014 loaded in the pool (for example)
* Login with a user that is linked to a player in 2014
* Go to week 13 update game scores page, delete all scores and uncheck all final (i.e. make week NOT_STARTED), click submit
* If lock_picks is set for 2014 week 13, you need to change it to None or some time in future
* The week results page should now show an enter picks link

**New Navbar**
I made a new week navbar to switch between pages more easily.

```Python
# major files
pick10/week_navbar.py  # this calculates what links to show up on a page
templates/pick10/week_navbar.html  # template for the week navbar
```

The links that show up depend on a number of factors.  Here is a few examples

*This is the pages a public user sees*
![public_user](https://cloud.githubusercontent.com/assets/1541768/9421933/14b00da2-484d-11e5-8536-e680c3d71b55.png)

*This is a user that is linked to a player*
![private_user](https://cloud.githubusercontent.com/assets/1541768/9421940/3a8a7e36-484d-11e5-9984-63cf52cdc473.png)

*This view has the enter picks link*
![enter_picks](https://cloud.githubusercontent.com/assets/1541768/9421942/493112ba-484d-11e5-84f9-f8d83a02876f.png)
